### DISCUSSION TITLE
Would help to update the contributing guide

### DISCUSSION THREAD

**Discussion by navaru** [parcel-bundler/parcel#4780] [General]
I wanted to try to fix a bug but got a bit stuck on how to start the development process.

For example for fixing bug #4772 how would you guys approach it?

**Comment by mischnic** [comment]
>  on how to start the development process.

What specifically are you having problems with?

Clone the repo, run `yarn`, copy the code sample into a new example in `packages/example/...` (temporarily for you to debug). Modify some code, run `yarn parcel build index.html` in the example folder to build.

(That issue you mentioned looks rather complicated to fix, even a core team member wasn't able to figure out the cause yet. We also have some easier issues marked with "help wanted" : https://github.com/parcel-bundler/parcel/issues?q=is%3Aopen+is%3Aissue+label%3A%22%E2%9C%A8+Parcel+2%22+label%3A%22%3Apray%3A+Help+Wanted%22)

**Comment by DeMoorJasper** [comment]
Hey @navaru that's a pretty complex bug to fix as your first issue, although I might be wrong.

Usually I create an example project in the examples folder that reproduces a bug or utilises a feature and then debug by writing console.logs mainly.

Sometimes I also start with writing an integration test and/or unit tests depending on the feature or bug.

For that particular issue you mentioned I started by writing an integration test and debugged it a little bit by starting in the HTMLPackager and working my way up to the bundlegraph where I assume the bug originates. However I wasn't able to figure it out entirely and created that issue to ensure my findings didn't go to waste and the bug would get fixed eventually.

As you can see in the mentioned branch in that issue I started with an integration test: https://github.com/parcel-bundler/parcel/tree/fix-html-bug

Like @mischnic mentioned we have labeled easier issues to get familiar with the codebase using [`Good First Issue`](https://github.com/parcel-bundler/parcel/issues?q=is%3Aopen+is%3Aissue+label%3A%22%E2%9C%A8+Parcel+2%22+label%3A%22%3Ababy%3A+Good+First+Issue%22) and [`Help Wanted`](https://github.com/parcel-bundler/parcel/issues?q=is%3Aopen+is%3Aissue+label%3A%22%E2%9C%A8+Parcel+2%22+label%3A%22%3Apray%3A+Help+Wanted%22)

**Comment by navaru** [comment]
I took an issue and here's my solution: https://github.com/navaru/parcel/commit/1cd8c95029526cfd333ac0227be53f79a64b66bb

Modified the `packages/examples/html` example locally, added a video, added the fix, what's next?

**Reply by DeMoorJasper** [reply]
Next step would be to open a Pull Request and await review and also ensure all tests still pass.

**Comment by navaru** [comment]
What are the rules on `yarn.lock`, if I add a new dependency to a package do I commit it too or?

What are the rules on versioning, I see in lerna is marked as "independent", do pull request expect version bumps?

I ask all these questions cause I want to update the CONTRIBUTING.md and make it clear.

**Reply by mischnic** [reply]
1. Yes, commit the changed lock file.

2. Don't bump versions of the parcel packages in PRs

**Comment by navaru** [comment]
I've created a draft for the contributing document, any feedback is appreciated, also for the Notes section please suggest anything that comes to mind.

https://github.com/navaru/parcel/blob/contributing-updates/CONTRIBUTING.md

**Reply by DeMoorJasper** [reply]
Looks great, feel free to open a Pull Request for that.

The only remark I have is that you removed a couple important sections like the section about testing, the test flow changed, but it should probably still be in there.

You also removed the section about `Environment variables, command line arguments` these are actually very important to ensure contributors understand that they need to disable cache while working on parcel as it will not invalidate without a version change and disabling workers is also useful for debugging although I've never needed it myself

**Reply by navaru** [reply]
I left those sections out cause I didn't know what was still relevant. I've updated the document.

**Reply by DeMoorJasper** [reply]
Great now only testing is missing and open a Pull Request please.

**Reply by navaru** [reply]
You're referring to the "Link to local parcel build" section? I don't see other mentions of testing.

**Reply by DeMoorJasper** [reply]
Thought there was a testing section but there was not.
Sorry about that, just open a PR, I'll merge it
